src/main/java/io/netty/buffer/api/Allocator.java

```diff
@@ -132,6 +132,15 @@ static void extend(Buf composite, Buf extension) {
         buf.extendWith(extension);
     }
 
+    /**
+     * Check if the given buffer is a {@linkplain #compose(Buf...) composite} buffer or not.
+     * @param composite The buffer to check.
+     * @return {@code true} if the given buffer was created with {@link #compose(Buf...)}, {@code false} otherwise.
+     */
+    static boolean isComposite(Buf composite) {
+        return composite.getClass() == CompositeBuf.class;
```

Comment:
Just a question... why not just `instanceof ` ?

Reply:
There are no sub-classes of CompositeBuf.

Reply:
sure... But I wondered if this is faster then instanceof ?

Reply:
It's certainly not slower, because it boils down to a pointer equality check, while `instanceof` may have to traverse the type hierarchy.

Reply:
ok... was just wondering as I usually just use instanceof.